inference-engine/thirdparty/clDNN/kernel_selector/core/common/kernel_selector_utils.cpp

```diff
@@ -199,20 +199,192 @@ std::vector<size_t> GetTensorFriendlyWorkGroups(const DataTensor& t) {
+                        priority_order[gws_idx] = axis_by_dims[layout_order[axis_idx]];
+                        break;
+                    }
+                }
+            }
+        }
+    };
+
+    auto one_layout = input_layout == output_layout;
+
+    auto simple_planar_layout = output_layout == DataLayout::bfyx || output_layout == DataLayout::yxfb ||
+                                output_layout == DataLayout::byxf || output_layout == DataLayout::fyxb ||
+                                output_layout == DataLayout::bfxy;
+
+    auto blocked_fsv_layout = output_layout == DataLayout::b_fs_yx_fsv4 || output_layout == DataLayout::fs_b_yx_fsv32 ||
+                              output_layout == DataLayout::b_fs_yx_fsv16 || output_layout == DataLayout::b_fs_zyx_fsv16 ||
+                              output_layout == DataLayout::b_fs_yx_fsv32 || output_layout == DataLayout::b_fs_zyx_fsv32;
+
+    auto blocked_bsv_fsv_layout = output_layout == DataLayout::bs_fs_yx_bsv16_fsv16 || output_layout == DataLayout::bs_fs_zyx_bsv16_fsv16;
+
+    auto try_change_priority_order = (simple_planar_layout || blocked_fsv_layout || blocked_bsv_fsv_layout) && one_layout;
+
+    if (try_change_priority_order) {
+        if (simple_planar_layout) {
+            switch (output_layout) {
+                case DataLayout::bfyx:
+                    layout_order = { x, y, f, b, z, w };
+                    break;
+                case DataLayout::yxfb:
+                    layout_order = { b, f, x, y, z, w };
+                    break;
+                case DataLayout::byxf:
+                    layout_order = { f, x, y, b, z, w };
+                    break;
+                case DataLayout::fyxb:
+                    layout_order = { b, x, y, f, z, w };
+                    break;
+                case DataLayout::bfxy:
+                    layout_order = { y, x, f, b, z, w };
+                    break;
+                case DataLayout::bfzyx:
+                    layout_order = { x, y, z, f, b, w };
+                    break;
+                case DataLayout::bfwzyx:
+                    layout_order = { x, y, z, w, f, b };
+                    break;
+                default:
+                    layout_order = { x, y, z, w, f, b };
+                    break;
+            }
+        } else if (blocked_fsv_layout) {
+            if (output_layout == DataLayout::b_fs_yx_fsv4 || output_layout == DataLayout::b_fs_yx_fsv16 || output_layout == DataLayout::b_fs_yx_fsv32)
+                layout_order = { f, x, y, b, z, w };
+            else if (output_layout == DataLayout::b_fs_zyx_fsv16 || output_layout == DataLayout::b_fs_zyx_fsv32)
+                layout_order = { f, x, y, z, b, w };
+            else // output_layout == DataLayout::fs_b_yx_fsv32
+                layout_order = { f, x, y, b, z, w };
+        } else if (blocked_bsv_fsv_layout) {
+            layout_order = { f, b, x, y, z, w };
+        }
+
+        if (try_change_priority_order)
```

Comment:
This condition is always true, since it is already under `if (try_change_priority_order)`